### PR TITLE
add cleanup function to navigation actions

### PR DESCRIPTION
**Pull request by Nikhil-Singhal-06**
# Issue number and link (if applicable)

Issue Number:

# Objective of pull request
Terminate running navigation if scenario shutdown.

# Pull request checklist

Your PR fulfills the following requirements:
- [ ] [Issue](https://github.com/IntelLabs/scenario_execution/issues) created that explains the change and why it's needed
- [ ] Tests are part of the PR (for bug fixes / features)
- [ ] [Docs](https://intellabs.github.io/scenario_execution/) reviewed and added / updated if needed (for bug fixes / features)
- [x] Format and Lint checks (`make check`) pass locally
- [x] PR applies Apache 2.0 License to all code files

# Pull request type

Please check your PR type:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation changes
- [ ] Other (please describe)

# What is the current behavior?
running navigation not terminating if scenario terminates.

# What is the new behavior?
navigation terminates when scenario shutdown.

# How to test

# Supplemental information

### CODE REVIEW

File: scenario_execution/scenario_execution/actions/nav_to_pose.py [hunk 1]
```diff
@@ -178,3 +178,7 @@ def cancel_task(self):
 
     def feedback_callback(self, msg):
         self.feedback = msg.feedback
+
+    def cleanup(self):
+        nav = BasicNavigator()
```
Comment (fred-labs) [inline scenario_execution/scenario_execution/actions/nav_to_pose.py:183]: have you tested this?
Is using the local `self.nav_to_pose_client` action client not working?

**Review by fred-labs — CHANGES_REQUESTED** [review]

File: scenario_execution/scenario_execution/actions/nav_to_pose.py (hunk 1 above)
Comment (Nikhil-Singhal-06) [reply scenario_execution/scenario_execution/actions/nav_to_pose.py:183]: Yes, I did. It is working.

However I didn't try using self.nav_to_pose_client

Comment (fred-labs) [reply scenario_execution/scenario_execution/actions/nav_to_pose.py:183]: please make use of self.nav_to_pose_client

**Review by fred-labs — CHANGES_REQUESTED** [review]

File: scenario_execution/scenario_execution/actions/nav_to_pose.py (hunk 1 above)
Comment (Nikhil-Singhal-06) [reply scenario_execution/scenario_execution/actions/nav_to_pose.py]: Done

**Review by fmirus — APPROVED** [review]

**Pull request merged**